### PR TITLE
Feat/hng 118 email template paginated table list

### LINKED ISSUE #118: [FEAT] Email Templates > SuperAdmin Email Management Dashboard > Paginated Table List of Template Items Component
## Description
Create a Paginated Table List of Template List Items Component for HTML Templates used in the Super Admin Email Management Dashboard for creating and editing email templates.

## Acceptance Criteria
- Reusable Paginated List of Template List Items Component for HTML Email Templates
- Component has Template List Item Component displayed as a Table List
- Component has a fluid width
- Component adheres to the brand colors and design

## Expected Outcome

- User should be able to view a list of HTML Templates
- User should be able to navigate through the list using pagination
- Responsive design of Component

## Purpose
As a User, I want to be able to view a paginated list of my Email HTML Templates

## Requirement

-  Makes use of Pagination Component
-  Makes use of Template List Item Component

## Testing

- [ ] test pagination
- [ ] test action callback `onPreview` is passed to list items

## Task
- [ ] Implement Reusable Paginated Template List Component
- [ ] Maps and displays Tabular list of Template List Item Components
- [ ] Display Pagination
- [ ] Pagination should navigate through the list of HTML Templates
- [ ] Pass `onPreview` prop from parent to children list item components
- [ ] Component adheres to the brand design and color scheme
- [ ] Component has a fluid width and is responsive across devices

## Deadline
You are given till Saturday 20th July 10:00AM WAT to complete the task.

## Picture
paginated templates

## Dev Mode Figma Link
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2319-9858&m=dev

### PR DESCRIPTION
**Pull request by buildwithhamzat**
# Description
I worked on the pagination component for email templates.

**Closes #118**

# Changes proposed

## What were you told to do?
[FEAT] Email Templates > SuperAdmin Email Management Dashboard > Paginated Table List of Template Items Component #118

## What did you do?
i created a new component where my pagination component reside
along side pagination template List and pagination item itself
they all have heir seperate folder inside the component/common folder
no dependancy was installed all logic was hardcoded

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
image
image
image

### CODE REVIEW

File: public/images/emailConfirmation.png
Comment (JulianaSau) [inline public/images/emailConfirmation.png:1]: what's this image for?

File: public/images/leftIcon.svg
Comment (JulianaSau) [inline public/images/leftIcon.svg:1]: i think for this you can just use an icon. instead of this image

File: public/images/rightIcon.svg
Comment (JulianaSau) [inline public/images/rightIcon.svg:1]: you can also just use an icon for this

File: src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    id: number;
+    name: string;
+    image: string;
+  }[];
+  onPreview: (id: number) => void;
+  itemsPerPage: number;
+}
+
+const PaginatedTemplateList = ({
+  templates,
```
Comment (JulianaSau) [inline src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx:20]: what happens if templates is undefined or is not an array?

**Review by JulianaSau — DISMISSED** [review]

File: public/images/emailConfirmation.png
Comment (buildwithhamzat) [reply public/images/emailConfirmation.png:1]: The image is a placeholder

File: public/images/leftIcon.svg
Comment (buildwithhamzat) [reply public/images/leftIcon.svg:1]: No icon library was installed that's why i decided to use the image provided in the figma

File: src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx (hunk 1 above)
Comment (buildwithhamzat) [reply src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx:20]: condition handled

File: src/components/common/Pagination/Pagination.test.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+
+import "@testing-library/jest-dom";
+
+import { describe, expect, it, vi } from "vitest";
```
Comment (Justfemi) [inline src/components/common/Pagination/Pagination.test.tsx:5]: Take out this line, it is available globally

File: src/app/guides/page.tsx [hunk 3]
```diff
@@ -8,11 +8,18 @@ import CustomButton from "~/components/common/Button/button";
...
+import templateTableList from "~/constants/templateList";
 import CharacterLimitTextarea from "../../components/common/CharacterLimitTextarea";
 
 const StyleGuide: React.FC = () => {
   const [text, setText] = useState("");
+  const [activePreview, setAcivePreview] = useState<number>(1);
+
+  const onPreview = (id: number) => {
+    setAcivePreview(id);
+  };
```
Comment (NerdPraise) [inline src/app/guides/page.tsx]: Unnecessary func
Just pass directly to component like `onPreview={(id) => setActivePreview(id)} `

File: src/app/guides/page.tsx [hunk 4]
```diff
@@ -8,11 +8,18 @@ import CustomButton from "~/components/common/Button/button";
...
 import { commentsData } from "~/components/common/Comment/ComentData";
 import CustomInput from "~/components/common/Input/input";
+import PaginatedTemplateList from "~/components/common/PaginatedTemplateList/PaginatedTemplateList";
 import Sidebar from "~/components/layouts/Sidebar";
+import templateTableList from "~/constants/templateList";
 import CharacterLimitTextarea from "../../components/common/CharacterLimitTextarea";
 
 const StyleGuide: React.FC = () => {
   const [text, setText] = useState("");
+  const [activePreview, setAcivePreview] = useState<number>(1);
```
Comment (NerdPraise) [inline src/app/guides/page.tsx]:
```suggestion
  const [activePreview, setActivePreview] = useState<number>(1);
```

File: src/app/guides/page.tsx [hunk 5]
```diff
@@ -667,6 +674,34 @@ const StyleGuide: React.FC = () => {
...
+            itemsPerPage<span className="text-pink-400">: </span>number;
+            <br />
+            onPreview<span className="text-pink-400">: </span>() =&gt; void;
+            <br />
+            templates: [];
+            <br />
+            activePreview<span className="text-pink-400">: </span>number;
+            <br />
+          </span>
+          <span className="block font-mono text-sm text-gray-100">{"}"}</span>
```
Comment (NerdPraise) [inline src/app/guides/page.tsx:743]: What is happening here with the squigly brackets?

File: src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  const currentTemplates = validTemplates.slice(
+    (currentPage - 1) * itemsPerPage,
+    currentPage * itemsPerPage,
+  );
+
+  return (
+    <div className="w-full overflow-x-auto">
+      <table className="min-w-full border-collapse">
+        <tbody>
+          {currentTemplates && currentTemplates.length > 0 ? (
```
Comment (NerdPraise) [inline src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx]:
```suggestion
          {currentTemplates.length > 0 ? (
```
The two checks are doing the practically same thing

File: src/app/guides/page.tsx (hunk 5 above)
Comment (buildwithhamzat) [reply src/app/guides/page.tsx:743]: I am trying to show the types required for the usage of the component. Used an approach from what we previously have in the file

File: src/components/common/Pagination/Pagination.test.tsx (hunk 2 above)
Comment (buildwithhamzat) [reply src/components/common/Pagination/Pagination.test.tsx:5]: I tried to check how it was done in other code but it seems nothing much was changed

Any pointer on that will be appreciated

File: public/images/leftIcon.svg
Comment (JulianaSau) [reply public/images/leftIcon.svg:1]: there's a library called lucide-icons

File: src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx (hunk 6 above)
Comment (JulianaSau) [reply src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx]: he should then add a null check if he removes this `currentTemplates`

Comment (buildwithhamzat) [reply src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx]: @JulianaSau This comment is outdated. I have handled the null check as well 


 ` const validTemplates = Array.isArray(templates) ? templates : [];`

```
 {currentTemplates.length > 0 ? (
            currentTemplates.map((template) => (
              <TemplateListItem
                key={template.id}
                template={template}
                onPreview={onPreview}
                activePreview={template.id === activePreview}
              />
            ))
          ) : (
            <tr>
              <td colSpan={3}>No templates available</td>
            </tr>
          )}
```

this is what we currently have in the PR you can check it to confirm

**Review by mrcoded — APPROVED** [review]

**Pull request merged**